### PR TITLE
Issues/7272 split self hosted flow

### PR DESCRIPTION
**Pull request by aerych**
Refs #7272 
This PR introduces the site address vc -- the first step in the self-hosted login flow.  The seconds step is the username/password form and will be implemented in a subsequent PR.
Requires #7375 to be merged first.

To test:
- Confirm the visual design was correctly implemented.
- Confirm the form is disabled until entering something resembling an URL (very loose rules used)
- Confirm that a valid WordPress url advances to the next form (currently the old self-hosted screen).
- Confirm that an invalid WordPress url prompts to double-check the address and try again.

Needs review: @nheagy 

Screen shots:

simulator screen shot jun 14 2017 8 45 01 pm
simulator screen shot jun 14 2017 8 45 30 pm
simulator screen shot jun 14 2017 8 45 38 pm

### CODE REVIEW

File: WordPress/Classes/ViewRelated/NUX/LoginSiteAddressViewController.swift [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        siteAddressHelpButton.setTitle(siteAddressHelpTitle, for: UIControlState())
+        siteAddressHelpButton.setTitle(siteAddressHelpTitle, for: .highlighted)
+        siteAddressHelpButton.titleLabel?.numberOfLines = 0
+    }
+
+
+    /// Configures the content of the text fields based on what is saved in `loginFields`.
+    ///
+    func configureTextFields() {
+        siteURLField.textInsets = UIEdgeInsetsMake(7, 20, 7, 20)
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginSiteAddressViewController.swift]: Whoops! Meant to update that.

File: WordPress/Classes/ViewRelated/NUX/LoginSiteAddressViewController.swift [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        configureSubmitButton(animating: loading)
+        navigationItem.hidesBackButton = loading
+    }
+
+
+    /// Configure the view for an editing state. Should only be called from viewWillAppear
+    /// as this method skips animating any change in height.
+    ///
+    func configureViewForEditingIfNeeded() {
+        // Check the helper to determine whether an editiing state should be assumed.
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/NUX/LoginSiteAddressViewController.swift]: 🤦‍♂️

**Comment by aerych** [comment]
Thanks Nate!  Feedback addressed.  Ready for another peek.

**Comment by nheagy** [comment]
Built and tested! 

:shipit:

**Comment by aerych** [comment]
Thanks @nheagy !!

**Pull request merged**